### ISSUE TITLE
Update package dependencies to a more recent version

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/msbuild#6774]
Microsoft.Build* packages reference quite old dependencies (runtime packages with a 4.* version). It would be great if msbuild would update their package dependencies to target the latest released 5.0.* runtime packages. The net benefit of doing so is a simplified dependency graph which results in not bringing in quite old packages which are hardly supported by dotnet/runtime (or corefx).

cc @rainersigwald @ericstj

**Comment by rainersigwald** [comment]
In general, this is nontrivial for us because we try to match the Visual Studio versions of the runtime packages. We looked at updating (https://github.com/dotnet/msbuild/pull/5910) and ran into several roadblocks (though honestly "we didn't have a super strong reason to do it" was probably the most salient factor).

Is the difference in package graph between the 4.x and 5.0 packages that significant? I don't see a way for us to get to the 6.0 ones since we'll need to ship simultaneously with 6.0 and don't want to add ourselves to the coherency problems.

**Comment by ViktorHofer** [comment]
> Is the difference in package graph between the 4.x and 5.0 packages that significant? I don't see a way for us to get to the 6.0 ones since we'll need to ship simultaneously with 6.0 and don't want to add ourselves to the coherency problems.

Let's take https://www.nuget.org/packages/System.Text.Encoding.CodePages/4.0.1 as an example which is referenced by msbuild. It doesn't provide a netstandard2.0 asset and hence pulls in all these old packages which we don't ship anymore for ages:

![image](https://user-images.githubusercontent.com/7412651/130247938-852d6752-3464-4a24-90f8-0c9648d11e9f.png)

There are many of such packages. Upgrading everything to 5.0.* guarantees that you use a coherent set of packages as they all shipped together in a single release.

>  I don't see a way for us to get to the 6.0 ones since we'll need to ship simultaneously with 6.0 and don't want to add ourselves to the coherency problems.

Right, if you don't want to be part of the coherency problem then you probably want to wait until 6.0.0 packages shipped. The 6.0.0 packages are again superior over the 5.0.* ones in many ways (much smaller, only include supported assets, trimmed dependencies, etc.).

**Comment by BenVillalobos** [comment]
This is blocked on https://github.com/dotnet/msbuild/issues/6807 and should be done soon after that issue, while it's fresh in our minds.

**Comment by rainersigwald** [comment]
Duplicate of #5852
